src/Faithlife.Analyzers/OverloadWithStringComparerAnalyzer.cs

```diff
@@ -0,0 +1,80 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Faithlife.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public sealed class OverloadWithStringComparerAnalyzer : DiagnosticAnalyzer
+	{
+		public override void Initialize(AnalysisContext context)
+		{
+			context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+			{
+				var enumerableType = compilationStartAnalysisContext.Compilation.GetTypeByMetadataName("System.Linq.Enumerable");
+				if (enumerableType != null)
+				{
+					compilationStartAnalysisContext.RegisterOperationAction(operationContext =>
+					{
+						var operation = (IInvocationOperation) operationContext.Operation;
+						var targetMethod = operation.TargetMethod;
+
+						if (targetMethod != null &&
+							targetMethod.ContainingType == enumerableType &&
+							(targetMethod.Name == "OrderBy" || targetMethod.Name == "OrderByDescending" || targetMethod.Name == "ThenBy" || targetMethod.Name == "ThenByDescending") &&
+							targetMethod.TypeArguments.Length == 2 &&
+							targetMethod.TypeArguments[1].SpecialType == SpecialType.System_String &&
+							targetMethod.Parameters.Length == 2)
+						{
+							operationContext.ReportDiagnostic(Diagnostic.Create(s_useStringComparerRule, GetMethodNameLocation(operation.Syntax)));
+						}
+					}, OperationKind.Invocation);
+				}
+			});
+		}
+
+		public const string UseStringComparerDiagnosticId = "FL0006";
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => s_rules;
+
+		private static Location GetMethodNameLocation(SyntaxNode syntaxNode)
```

Comment:
Didn't want to improve `OperationExtensions` (it'd be fine to rename it to `SyntaxExtensions`)?

Reply:
Oh, never mind; extracting that method for reuse is in my unmerged PR.